Modules/Core/SpatialObjects/include/itkImageMaskSpatialObject.hxx

```diff
@@ -71,77 +71,48 @@ ImageMaskSpatialObject< TDimension, TPixel >
   return false;
 }
 
+
 template< unsigned int TDimension, typename TPixel >
 void
 ImageMaskSpatialObject< TDimension, TPixel >
 ::ComputeMyBoundingBox()
 {
-  using IteratorType = ImageRegionConstIteratorWithIndex< ImageType >;
-  IteratorType it( this->GetImage(),
-    this->GetImage()->GetLargestPossibleRegion() );
-  IteratorType prevIt( this->GetImage(),
-    this->GetImage()->GetLargestPossibleRegion() );
-  it.GoToBegin();
-  prevIt = it;
-
-  bool first = true;
-  PixelType outsideValue = NumericTraits< PixelType >::ZeroValue();
-  PixelType value = outsideValue;
-  PixelType prevValue = outsideValue;
-  IndexType tmpIndex;
-  PointType tmpPoint;
-  int count = 0;
-  int rowSize
-    = this->GetImage()->GetLargestPossibleRegion().GetSize()[0];
-  while ( !it.IsAtEnd() )
-    {
-    value = it.Get();
-    if ( value != prevValue || ( count == rowSize-1 && value != outsideValue ) )
-      {
-      prevValue = value;
-      if( value == outsideValue )
-        {
-        tmpIndex = prevIt.GetIndex();
-        }
-      else
-        {
-        tmpIndex = it.GetIndex();
-        }
-      this->GetImage()->TransformIndexToPhysicalPoint( tmpIndex, tmpPoint );
-      if( first )
-        {
-        first = false;
-        this->GetModifiableMyBoundingBoxInObjectSpace()->SetMinimum( tmpPoint );
-        this->GetModifiableMyBoundingBoxInObjectSpace()->SetMaximum( tmpPoint );
-        }
-      else
-        {
-        this->GetModifiableMyBoundingBoxInObjectSpace()->ConsiderPoint( tmpPoint );
-        }
-      }
-    prevIt = it;
-    ++it;
-    ++count;
-    if( count == rowSize )
-      {
-      count = 0;
-      prevValue = outsideValue;
-      }
-    }
+  const ImageType* const image = this->GetImage();
+  assert(image != nullptr);
```

Comment:
Just a detail on style here. I presume that this method is callable without setting the classes Image member. So its possible a usage error could get to this location in Release mode. 

This should be ` itkAssertInDebugOrThrowInReleaseMacro` to ensure safety in release mode too.

Reply:
Another possibility could be to just "do nothing" and return, when `image == nullptr`. Note that the original code just used the image pointer without any null-checking.